api/edge_api/identities/views.py

```diff
@@ -197,8 +199,18 @@ def initial(self, request, *args, **kwargs):
             != self.kwargs["environment_api_key"]
         ):
             raise PermissionDenied("Identity does not belong to this environment.")
+        identity = build_identity_model(identity_document)
 
-        self.identity = build_identity_model(identity_document)
+        valid_feature_names = set(
+            FeatureState.objects.filter(
+                environment__api_key=identity.environment_api_key
+            ).values_list("feature__name", flat=True)
+        )
+        identity_feature_names = {fs.feature.name for fs in identity.identity_features}
+        if not identity_feature_names.issubset(valid_feature_names):
+            identity.prune_features(valid_feature_names)
+            sync_identity_document.delay(args=(build_identity_dict(identity),))
```

Comment:
I think we'd be better off just sending the identity id or composite key (depending on what we have available here )to the task and grabbing the identity out of the database in the task. That should avoid any issues with serialization of task arguments (and remove the need for the DjangoJSON encoder that you added). 

It adds another read to the database in the task code but I think it will be cleaner overall.

Reply:
Yeah, that makes sense